index.js

```diff
@@ -25,15 +25,18 @@ function querystring(query) {
     , result = {}
     , part;
 
-  //
-  // Little nifty parsing hack, leverage the fact that RegExp.exec increments
-  // the lastIndex property so we can continue executing this loop until we've
-  // parsed all results.
-  //
-  for (;
-    part = parser.exec(query);
-    result[decode(part[1])] = decode(part[2])
-  );
+  while (part = parser.exec(query)) {
+    var key = decode(part[1])
+      , value = decode(part[2]);
+
+    //
+    // Prevent overriding of existing properties. This ensures that build-in
+    // methods like `toString` or __proto__ are not overriden by malicious
+    // querystrings.
+    //
+    if (key in result) continue;
```

Comment:
We should be safe even without this, as all of them can be safely overridden, with the exception of `__proto__` but setting it to a string has no effect.

That said I'm also ok with this.

Reply:
Without the `key in result` check, the test will fail because toString() can be overridden.

Reply:
Yes, that's my point, if your query string has a parameter called `toString` the expected behaviour is probably to find that parameter in the parsed object.

It overrides the `toString` inherited by `Object.prototype` but that another issue. I mean ideally the parsed object should not inherit from `Object.prototype` at all but we can't use `Object.create(null)` due to old browsers support.